### DISCUSSION TITLE
Lesson 12: Deploying DecentralizedStableCoin

### DISCUSSION THREAD

**Discussion by ghost** [Cyfrin/foundry-full-course-cu#2123] [Q&A]
Hello!

So, as I am writing the contracts from lesson 12 about Stablecoin, I had to change the constructor of DecentralizedStableCoin a little because of the Ownable to this: 
`constructor() ERC20("DecentralizedStableCoin", "DSC") Ownable(address(msg.sender)) { }`. 
As you can see I passed msg.sender (the contract that is going to call my DecentralizedStableCoin). I checked which one is that and I saw that the address of the contract calling it is my MetaMask address from the Setting Up MetaMask for Localhost/Anvil Network. 

If I am understanding the whole thing correctly: through the run function in the Deploy script with the help of the HelperConfig contract I am:
 1. setting the active network
 3. deploying my contracts DecentralizedStableCoin, DSCEngine 
 And without the need to run Anvil in the terminal, my MetaMask calls/deploys my contract (DSCEngine) only because I created the settings for Anvil network and because of the run function with its vm.startBroadcast (which executes ok no matter if I pass the DEFAULT_ANVIL_PRIVATE_KEY or not as deployerKey: `vm.startBroadcast(deployerKey)`)

Could someone with an understanding of the whole process please confirm my understanding or explain it to me if I am wrong?

Thank You 🙏

**Comment by EngrPips** [comment]
It seems like you figured this out. Keep crushing it.
